app/models/user.rb

```diff
@@ -47,6 +47,8 @@ class User < ActiveRecord::Base
   scope :by_document,    -> (document_type, document_number) { where(document_type: document_type, document_number: document_number) }
 
   before_validation :clean_document_number
+  
+  before_save :check_if_confirmation
```

Comment:
I'm not sure we need this check on every user's save. It's probably enough to do it only after first registration, or email verification and email address change.